keyboards/e6v2/le/keymaps/johu/keymap.c

```diff
@@ -0,0 +1,101 @@
+#include QMK_KEYBOARD_H
+
+#define _BL 0
+#define _FL 1
+#define _SL 2
+#define _CL 3
+
+// Tap dance defines
+#define CTL_ESC     CTL_T(KC_ESC)               // Tap for Esc, hold for Ctrl
+#define HPR_TAB     ALL_T(KC_TAB)               // Tap for Tab, hold for Hyper (Super+Ctrl+Shift+Alt)
+#define SFT_ENT     SFT_T(KC_ENT)               // Tap for Enter, hold for Shift
+#define KC_SPFN     LT(_SL, KC_SPC)             // Tap for space, hold for function layer (aka SpaceFN)
+
+// Helpers
+#define ______ KC_TRNS
```

Comment:
QMK has a built-in keycode to do this automatically.

https://github.com/qmk/qmk_firmware/blob/27e216fc1a481bb3e65832458bd2b199cb499e0b/quantum/quantum_keycodes.h#L29-L31

It's much preferred to use the built-in code.

The built-in code is seven underscores, rather than six as you have here.

Reply:
When the define is dropped the compile fails because it can't not find the define. Is there another required include that I am overseen?

Reply:
If you delete Lines 14-15 and then change all the instances of  
`______` (6 underscores) to  
`_______` (7 underscores), it will work.

The file I linked before is included by using `#include QMK_KEYBOARD_H`, which is already present in your keymap.

Reply:
Thanks.